core/frontend/src/tile/IndexedDBCache.ts

```diff
@@ -0,0 +1,185 @@
+
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+/** @beta */
+export interface ILocalCache {
+  fetch(url: string, callback: (uniqueId: string) => Promise<Response>, callBackUrl?: string): Promise<ArrayBuffer>;
+}
+
+/** @beta */
+export class PassThroughCache implements ILocalCache {
+  public async fetch(uniqueId: string, callback: (url: string) => Promise<Response>, callBackUrl?: string): Promise<ArrayBuffer> {
+    if (callBackUrl) {
+      return (await callback(callBackUrl)).arrayBuffer();
+    }
+    return (await callback(uniqueId)).arrayBuffer();
+  }
+}
+
+/** @beta */
+export class IndexedDBCache implements ILocalCache{
+  private _db: any;
+  private _dbName: string = "IDB";
+  private _expirationTime?: number;
+
+  public constructor(dbName: string, expirationTime?: number) {
+    this._dbName = dbName;
+    if (expirationTime) {
+      this._expirationTime = expirationTime;
+    }
+  }
+
+  protected async open(dbCache: IndexedDBCache){
```

Comment:
This is an instance method, why does it take an IndexedDBCache argument instead of operating on `this`?